### PR TITLE
Set the default WSL starting directory to ~

### PR DESCRIPTION
**Pull request by DHowett**
The update that enables this on 20H1+, [KB5007253], went out in November
2021.

[KB5007253]: https://support.microsoft.com/en-us/topic/november-22-2021-kb5007253-os-builds-19041-1387-19042-1387-19043-1387-and-19044-1387-preview-d1847be9-46c1-49fc-bf56-1d469fc1b3af

### CODE REVIEW

**Comment by DHowett** [comment]
image
image

_@PankajBhojwani what is that at the top!_

**Comment by PankajBhojwani** [comment]
> what is that at the top!

That's wild, do you have any repro steps?

File: src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp [hunk 1]
```diff
@@ -23,6 +24,18 @@ static constexpr wchar_t RegKeyDistroName[] = L"DistributionName";
...
+static bool isWslDashDashCdAvailableForLinuxPaths() noexcept
+{
+    OSVERSIONINFOEXW osver{};
+    osver.dwOSVersionInfoSize = sizeof(osver);
+    osver.dwBuildNumber = 19041;
+
+    DWORDLONG dwlConditionMask = 0;
+    VER_SET_CONDITION(dwlConditionMask, VER_BUILDNUMBER, VER_GREATER_EQUAL);
+
+    return VerifyVersionInfoW(&osver, VER_BUILDNUMBER, dwlConditionMask) != FALSE;
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp:36]: i know it seems weird to check this when Terminal's min version is 19041 already, but... 1.12 still runs on 18362 and this is  backport candidate. I don't want to blow up folks on that version combo

**Review by carlos-zamora — APPROVED** [review]

**Comment by DHowett** [comment]
@msftbot make sure @zadjii-msft signs off

**Review by lhecker — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]

**Comment by DHowett** [comment]
@msftbot merge this in 1 minute

**Pull request merged**